scripts/cmake/vcpkg_copy_tools.cmake

```
@@ -53,6 +54,18 @@ function(vcpkg_copy_tools)
         if(EXISTS "${tool_pdb}")
             file(COPY "${tool_pdb}" DESTINATION "${CURRENT_PACKAGES_DIR}/tools/${PORT}")
         endif()
+
+        if(VCPKG_TARGET_IS_LINUX AND ("${VCPKG_LIBRARY_LINKAGE}" STREQUAL "dynamic"))
+            find_program(PATCHELF_FOUND NAMES patchelf)
+            if(PATCHELF_FOUND)
+                # Fix RPATH: works but requires patchelf

```

This "optimistic" behavior which falls back if `patchelf` isn't found breaks binary caching, since you get a meaningfully different result on a system with `patchelf` vs a system without. If you want to do this it needs to fail in the event `patchelf` isn't found rather than silently doing nothing.

---

Ok, I forgot the binary caching and I only wanted to keep the existing behavior on systems that do not have `patchelf`. I will make the required modification and move this code to `vcpkg_copy_tool_dependencies`. Anyway, if it is acceptable to "force" people to install `patchelf`, you are right, it is far more coherent to fail with a meaningful error than silently do nothing.

Thanks for your review.

---

> I only wanted to keep the existing behavior on systems that do not have `patchelf`.

If you want to provide a "I don't have patchelf" alternative you can make it a triplet setting and turn it on in the `x64-linux-dynamic` triplet.

---

well, I'm wondering if it's anyway a good thing that something can be built, but remains unusable without setting LD_LIBRARY_PATH or patching anyway the rpath. Maybe the sane way of viewing things is to make patchelf a prerequisite. BTW, `x64-linux-dynamic` must still be created by users, no ? Is it scheduled to add it permanently to "community" folder in VCPKG distribution ?